### ISSUE TITLE
[Doc] Reset Camera Position

### ISSUE THREAD

**Issue by Dev-Lan** [visgl/deck.gl#8198]
### Link

https://deck.gl/docs/developer-guide/interactivity

### Description

I am using deckgl to view images of cells taken with a microscope. Basic built-in camera controls are sufficient for my use case. However, I would like to reset the camera position. There is a section in the docs titled "Reset Camera Position", however, I'm running into an issue and am unsure if it's a bug or a miscommunication in the docs.

Resetting the `initialViewState` works, but only once. That is,

1. User pans/zooms image.
2. User clicks "reset" button. WORKS!
3. User pans/zooms image again.
4. User clicks "reset" button again. Does nothing :'(

Now if the object passed into the initialViewState changes, it works again, so it seems this approach only triggers an update if the `initialViewState` changes.

The reason I have submitted this to docs initially is that based on [this discussion](https://github.com/visgl/deck.gl/discussions/7023), it seems like maybe setting the initialViewState isn't intended as a way to reset the view.

Any clarification here would be much appreciated. Thanks!

**Comment by Dev-Lan** [comment]
Similar to the discussion, if I add small amount of noise to my target the behavior works as I want, but feels like a hack.

```
function resetView() {
    deckgl.setProps({
        initialViewState: {
            zoom: 0,
            bearing: 0,
            target: [
                width / 2 + Math.random() * 0.001,
                height / 2 + Math.random() * 0.001,
                0,
            ],
        },
    });
}
```

**Comment by Pessimistress** [comment]
The way it currently works, if the new value supplied to `initialViewState` is deeply equal to the old value, it is considered "not changed" and ignored. I agree that it is inconvenient in this use case. Maybe we should change it to shallow comparison in the next version? @ibgreen @chrisgervang

**Comment by ibgreen** [comment]
>  Maybe we should change it to shallow comparison in the next version?

Yes I could get behind that. It would certainly give the user more options, but also require more care to avoid triggering a reset (like when using react with a fixed array as initial state). So we might break some existing apps...

Alternatively adding new props but this does not seem very attractive
-  `initialViewStateShallow`
- `initialViewStateCompare: 'deep' | 'shallow'`
- ...

**Comment by chrisgervang** [comment]
It does seem inconvenient that it only resets if the view is different. A shallow comparison should work, but would also reset when the object is spread or copied.

**Comment by ibgreen** [comment]
Yes, effectively the app would now be using spread or copying as a technique to trigger the reset.

**Comment by chrisgervang** [comment]
Don't react/redux applications rely on copying even when the values haven't changed?

**Comment by chrisgervang** [comment]
`initialViewStateShallow` looks good to me if it's just this prop.

If deck is tracking changes for other props maybe an object to configure them all..

```
comparisons: {
  initialViewState: "shallow"
  initialEffects ...
}
```

**Comment by Dev-Lan** [comment]
I just implemented a camera reset function in a new view and was surprised again why it wasn't working 😂

Thankfully I had a vague memory of writing an issue related to cameras in deckgl.

**Comment by chrisgervang** [comment]
Since this is only a behavior on initialViewState, what do we think about adding a property to the viewState to configure this behavior?

```
initialViewState: {
  ...,
  longitude: 90,
  compare: 'shallow' | 'deep'
}
```
